rama-http/src/service/web/endpoint/response/sse.rs

```diff
@@ -0,0 +1,262 @@
+//! Server-Sent Events (SSE) response.
+
+use futures_lite::Stream;
+use futures_util::TryStream;
+use rama_core::error::BoxError;
+use rama_http_headers::{CacheControl, Connection, ContentType};
+use rama_http_types::{
+    Body, Response,
+    sse::{
+        Event, EventDataWrite,
+        server::{KeepAlive, KeepAliveStream, SseResponseBody},
+    },
+};
+use std::fmt;
+
+use super::{Headers, IntoResponse};
+
+/// An SSE response
+#[must_use]
+pub struct Sse<S> {
+    stream: S,
+}
+
+impl<S> Sse<S> {
+    /// Create a new [`Sse`] response that will respond with the given stream of
+    /// [`Event`]s.
+    pub fn new<T>(stream: S) -> Self
+    where
+        S: TryStream<Ok = Event<T>> + Send + 'static,
+        S::Error: Into<BoxError>,
+        T: EventDataWrite,
+    {
+        Sse { stream }
+    }
+
+    /// Configure the interval between keep-alive messages.
+    ///
+    /// Defaults to no keep-alive messages.
+    pub fn keep_alive<T, E>(self, keep_alive: KeepAlive<T>) -> Sse<KeepAliveStream<S, T>>
```

Comment:
```suggestion
    pub fn with_keep_alive<T, E>(self, keep_alive: KeepAlive<T>) -> Sse<KeepAliveStream<S, T>>
```
We might aswel be consistent everywhere